packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,540 @@
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingStart")}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send bufferingStart event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendBufferingUpdate(int position) {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingUpdate")},
+        {flutter::EncodableValue("values"), flutter::EncodableValue(position)}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send bufferingUpdate event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendBufferingEnd() {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingEnd")}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send bufferingEnd event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::OnPrepared(void *data) {
+  VideoPlayer *player = reinterpret_cast<VideoPlayer *>(data);
+  LOG_INFO("[VideoPlayer] video player is prepared");
+  if (!player->is_initialized_) {
+    player->SendInitialized();
+  }
+}
+
+void VideoPlayer::OnBuffering(int percent, void *data) {
+  LOG_INFO("[VideoPlayer] percent: %d", percent);
+  VideoPlayer *player = reinterpret_cast<VideoPlayer *>(data);
+  if (percent == 100) {
+    player->SendBufferingEnd();
+    player->is_buffering_ = false;
+  } else if (!player->is_buffering_ && percent <= 5) {
+    player->SendBufferingStart();
+    player->is_buffering_ = true;
+  } else {
+    player->SendBufferingUpdate(percent);
+  }
+}
+
+void VideoPlayer::OnSeekCompleted(void *data) {
+  VideoPlayer *player = reinterpret_cast<VideoPlayer *>(data);
+  LOG_INFO("[VideoPlayer] completed to seek");
+  if (player->on_seek_completed_) {
+    player->on_seek_completed_();
+    player->on_seek_completed_ = nullptr;
+  }
+}
+
+void VideoPlayer::OnPlayCompleted(void *data) {
+  VideoPlayer *player = reinterpret_cast<VideoPlayer *>(data);
+  LOG_INFO("[VideoPlayer] completed to play video");
+  if (player->event_sink_) {
+    flutter::EncodableMap encodables = {{flutter::EncodableValue("event"),
+                                         flutter::EncodableValue("completed")}};
+    flutter::EncodableValue eventValue(encodables);
+    player->event_sink_->Success(eventValue);
+    player->Pause();
+  }
+}
+
+void VideoPlayer::OnError(int error_code, void *user_data) {
+  VideoPlayer *player = reinterpret_cast<VideoPlayer *>(user_data);
+  LOG_ERROR("[VideoPlayer] error code: %s", get_error_message(error_code));
+  if (player->event_sink_) {
+    player->event_sink_->Error("Video player had error",
+                               get_error_message(error_code));
```

Comment:
Changing this to

```suggestion
    player->event_sink_->Error(
        "Player error", std::string("Error: ") + get_error_message(code));
```
may fix the problem in https://github.com/flutter-tizen/flutter-tizen/issues/217#issuecomment-1328270433.

FYI) We had a similar problem which has been fixed by https://github.com/flutter-tizen/plugins/pull/262, but in your case the message returned by `get_error_message` is an empty string so you need some extra processing.

Reply:
fixed